### PR TITLE
Some lambda fixes

### PR DESCRIPTION
**Pull request by akva2**
- Remove an unused capture
- Restructure to avoid a capture that causes warnings in clang but is required by g++-14

### CODE REVIEW

**Review by atgeirr — APPROVED** [review]

File: opm/simulators/flow/FlowProblemBlackoil.hpp [hunk 1]
```diff
@@ -1524,7 +1524,7 @@ class FlowProblemBlackoil : public FlowProblem<TypeTag>
         using ViolationLevel = typename Satfunc::PhaseChecks::
             SatfuncConsistencyCheckManager<Scalar>::ViolationLevel;
 
-        auto reportFailures = [&sfuncConsistencyChecks, this]
+        auto reportFailures = [&sfuncConsistencyChecks]
```
Comment (bska) [inline opm/simulators/flow/FlowProblemBlackoil.hpp:1527]: Thanks!  I forgot to remove `this` in one of the later updates here.

**Review by bska — APPROVED** [review]
Looks good to me.  Please merge when green.

**Comment by akva2** [comment]
The second commit here fixes a build error with g++-14 so I think it should be backported.

**Comment by bska** [comment]
> The second commit here fixes a build error with g++-14 so I think it should be backported.

Very good.  I'll port it now.

**Pull request merged**